markdown-export/render-rules.ts

```diff
@@ -16,6 +16,44 @@ interface TableCell {
     align: string | undefined;
 }
 
+function parseAndFormatDate(str: string): string {
+    // TW date format (spaces added for clarity): [UTC] YYYY 0MM 0DD 0hh 0mm 0ss 0XXX
+    const regex = /^(\d{4})(\d{2})(\d{2})(\d{2})(\d{2})(\d{2})(\d{3})$/;
+    const match = str.match(regex);
+    if (match) {
+        const isoDateString = `${match[1]}-${match[2]}-${match[3]}T${match[4]}:${match[5]}:${match[6]}Z`;
+        const date = new Date(isoDateString);
+        return date.toISOString();
+    } else {
+        console.warn("Invalid date format", str);
+        return str;
+    }
+}
+
+/** Field values are converted to strings by TW, we can switch them back */
+function formatYAMLString(fieldValue: any, enableNumbers : boolean = true): string {
+    const datePatternTW = /^\d{17}$/;
+    if (fieldValue.toISOString) {
+        fieldValue = "'" + fieldValue.toISOString() + "'";
+    }
+    else if (datePatternTW.test(fieldValue)) {
```

Comment:
Use the same "date pattern" here as inside the parse function, since the parse function doesn't accept all `^\d{17}$` strings.

Reply:
Pulled into a single `const` but no longer need the second function so have left it in the function where it is used. LMK if you agree.